crates/ruff_linter/src/checkers/ast/analyze/expression.rs

```diff
@@ -1160,7 +1160,7 @@ pub(crate) fn expression(expr: &Expr, checker: &mut Checker) {
                     }
                 }
                 if checker.enabled(Rule::PrintfStringFormatting) {
-                    pyupgrade::rules::printf_string_formatting(checker, format_string, right);
+                    pyupgrade::rules::printf_string_formatting(checker, bin_op, format_string);
```

Comment:
The function requires the range and the right side of the binary expression so we let's just pass the binary expression as a whole.